common/source/nrt_fpga_device.cpp

```diff
@@ -0,0 +1,183 @@
+#include "nrt_fpga_device.h"
+
+#include <thread>
+
+NRTFPGADevice::NRTFPGADevice() :
+	nrt_fpga_fd_(0),
+	nrt_fpga_thread_([&](){ this->Work(); }) {
+	CommonKeyPairs config = {
+		{"fpga_device_file", "/dev/iofpga"}
+	};
+	ReConfig(config);
+	if (ReConfig(config) != kNoError)
+		return;
+	nrt_fpga_thread_.Start();
+}
+
+NRTFPGADevice::~NRTFPGADevice() {
+	DeConfig();
+}
+
+error_condition	NRTFPGADevice::ReConfig(const CommonKeyPairs& config) {
+
+	auto it_device_file = config.find(kFPGADeviceFile);
+	if (it_device_file == config.end())
+		return make_error_condition(ErrorCode::kDeviceFileNotSpecified);
+
+	int fpga_device_fd = open(it_device_file->second.c_str(), O_RDWR | O_NONBLOCK);
+	if (fpga_device_fd < 0)
+		return make_error_condition(ErrorCode::kOpenFileError);
+
+	error_condition result = kNoError;
+	// Select configuration page, prepare to configure FPGA
+	ioctl(fpga_device_fd, IOFPGA_SEL_CONF, NULL);
```

Comment:
Before setting configurations, we could get informantion from "info" page to calculate register offsets, I'll send you the example.